src/wp-includes/class-wp-block.php

```diff
@@ -446,6 +455,12 @@ public function render( $options = array() ) {
 			}
 		}
 
+		if ( ! empty( $computed_attributes ) ) {
```

Comment:
That should ensure it only applies changes if the block's content is not empty.

```suggestion
		if ( ! empty( $computed_attributes ) && ! empty( $block_content ) ) {
```